contracts/ArrowToken.sol

```diff
@@ -1,29 +1,37 @@
-pragma solidity ^0.8.11;
-
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.11;
 
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 /**
-    @title Arrow Token implementation
- */
-contract ArrowToken is ERC20, Ownable {
+    @title Arrow Token implementation that is upgradeable
+    @notice Create ARROW tokens of an `initialSupply` amount
+    @dev If we were to upgrade the initializer by introducing new variables, we need to create a normal function like `initializeV2` (https://forum.openzeppelin.com/t/uups-proxies-tutorial-solidity-javascript/7786/53), and make sure to include all existing variables from the current version of the contract to avoid storage collision
+*/
+contract ArrowToken is
+    Initializable,
+    ERC20Upgradeable,
+    OwnableUpgradeable,
+    UUPSUpgradeable
+{
+    // implement the UUPS interface
+    function _authorizeUpgrade(address) internal override onlyOwner {}
 
-    /// @notice This function assigns msg.sender as the token admin and mints them initial supply
-    constructor(uint256 initialSupply, string memory _name, string memory _symbol)
-        ERC20(_name, _symbol)
-    {
-        _mint(msg.sender, initialSupply);
-    }
```

Comment:
I think we just need to include the below codes rather than the explicit constructor.
```
/// @custom:oz-upgrades-unsafe-allow constructor
constructor() initializer {}
```
https://forum.openzeppelin.com/t/security-advisory-initialize-uups-implementation-contracts/15301

Reply:
Yeah, that's what I was referring to. We need a constructor that initialises the contract.